### ISSUE TITLE
Error in loading Pretrained model

### ISSUE THREAD

**Issue by tusharkumar91** [lichengunc/MAttNet#24]
Hello Licheng,

To use the pretrained model for refcoco unc split, I download it from [here](http://bvision.cs.unc.edu/licheng/MattNet/pretrained/refcoco_unc.zip). But when I attempt to run the evaluation with this pretrained model, I get the following error while running the evaluation script.

**Error**

```Traceback (most recent call last):
  File "./tools/eval_easy.py", line 105, in <module>
    evaluate(params)
  File "./tools/eval_easy.py", line 46, in evaluate
    model = load_model(model_path, model_opt)
  File "./tools/eval_easy.py", line 29, in load_model
    model.load_state_dict(checkpoint['model'].state_dict())
  File "/home/tusharkumar91/anaconda3/envs/py2/lib/python2.7/site-packages/torch/nn/modules/module.py", line 719, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for JointMatching:
        Missing key(s) in state_dict: "sub_encoder.att_fuse.1.num_batches_tracked", "sub_matching.vis_emb_fc.1.num_batches_tracked", "sub_matching.vis_emb_fc.5.num_batc
hes_tracked", "sub_matching.lang_emb_fc.1.num_batches_tracked", "sub_matching.lang_emb_fc.5.num_batches_tracked", "loc_matching.vis_emb_fc.1.num_batches_tracked", "loc_
matching.vis_emb_fc.5.num_batches_tracked", "loc_matching.lang_emb_fc.1.num_batches_tracked", "loc_matching.lang_emb_fc.5.num_batches_tracked", "rel_matching.vis_emb_fc
.1.num_batches_tracked", "rel_matching.vis_emb_fc.5.num_batches_tracked", "rel_matching.lang_emb_fc.1.num_batches_tracked", "rel_matching.lang_emb_fc.5.num_batches_trac
ked". 
```

**Command used for evaluation**

```
./experiments/scripts/eval_easy.sh GPUID refcoco unc
```

However, if I train MAttnet for just one iteration and use the model saved at that checkpoint, the evaluation runs just fine without any issues.

I was wondering if I am missing something here or is the link that I used to download the pretrained model incorrect ?

Thanks for your help,
Tushar

**Comment by dami23** [comment]
The MAttNet extracts RoI features through mask-faster-rcnn which is built by python 2.7. So, please use python 2.7 + pytorch 0.4.1, you can get the right output. Another issue is that, if you want to train a model, adopt python 2.7 + pytorch 0.4.1, there are some compatible problems with python 3.5 and pytorch 1.0.

**Comment by tusharkumar91** [comment]
Hello @dami23 ,

Thank you for your response.
However, I did notice this requirement regarding python 2.7 and the issue that I am getting is indeed with python 2.7 and pytorch 0.4.1.
I believe all the dependencies are fine because, I am able to train the network for an epoch, it is upon using the pre-trained weights that I face this issue regarding incompatible dictionary structure.

Thanks,
Tushar

**Comment by dami23** [comment]
Hello @tusharkumar91,
the line 211 in train.py is not compatible with pytorch 0.4.1, I do not still find a resolution. Maybe you also have a problem with this line. But, with python 2.7 +pytorch0.4.1, I get the right output when I run ./tools/eval_easy.py.

Best,
mi

**Comment by tusharkumar91** [comment]
@dami23, Yes, I was facing that error also but my error is not related to that. My issue is I just could not use the pre-trained model weights and run the evaluation scripts(without any training). 

Could you please let me know that when you were running the eval_easy script, were you using the pre-trained weights provided in the README file. And for which dataset/split were you running them?

Thanks,
Tushar

**Comment by dami23** [comment]
Hi, @tusharkumar91, the pretrained model cannot be loaded correctly, the error is the same as yours. My colleague and I checked the code thoroughly, but we did not find any fault in the codes and we also did not get a resolution. But, with python 2.7 + pytorch0.4.1, you can train a model yourself. 
I've contacted Licheng, he said he cannot reproduce the error, so he cannot give any suggestions.

**Comment by tusharkumar91** [comment]
I got the issue fixed. It was actually the pytorch version. 
It seems that after downgrading my pytorch to 0.3, the pre-trained model worked fine after that and I was able to get the same numbers reported in the ReadMe for localization tasks.

When trying to execute extract_mrcn_det_feats.py in order to evaluate segmentation masks, I get the same error as this [issue](https://github.com/lichengunc/MAttNet/issues/4), which I believe would be fixed if I downgrade pytorch to 0.2.1.

Thank you @dami23  for your help on this.
